doc/examples/motion_planning_python_api/scripts/motion_planning_python_api_tutorial.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+"""
+A script to outline the fundamentals of the moveit_py motion planning API.
+"""
+import numpy as np
+import rclpy
+from rclpy.node import Node
+import sys
+import threading
+import time
+
+from geometry_msgs.msg import Pose, PoseStamped
+from moveit_msgs.msg import Constraints
+
+from moveit.planning import (
+    MoveItPy,
+    PlanRequestParameters,
+    MultiPipelinePlanRequestParameters,
+)
+from moveit.core.robot_state import RobotState
+from moveit.core.kinematic_constraints import construct_joint_constraint
+
+from ament_index_python.packages import get_package_share_directory
+from moveit_configs_utils import MoveItConfigsBuilder
+import yaml
+
+
+def main():
+
+    ###################################################################
+    # MoveItPy Setup
+    ###################################################################
+    rclpy.init()
+    logger = rclpy.logging.get_logger("moveit_py.pose_goal")
+
+    # instantiate MoveItPy instance and get planning component
+    moveit = MoveItPy(node_name="moveit_py")
+    panda_arm = moveit.get_planning_component("panda_arm")
+    logger.info("MoveItPy instance created")
+
+    ###########################################################################
+    # Plan 1 - set states with predefined string
+    ###########################################################################
+
+    # set plan start state using predefined state
+    panda_arm.set_start_state(configuration_name="ready")
+
+    # set pose goal using predefined state
+    panda_arm.set_goal_state(configuration_name="extended")
+
+    # plan to goal
+    logger.info("Planning trajectory")
+    plan_result = panda_arm.plan()
+
+    # execute the plan
+    if plan_result:
+        logger.info("Executing plan")
+        robot_trajectory = plan_result.trajectory
+        moveit.execute(robot_trajectory, blocking=True, controllers=[])
```

Comment:
This chunk of code is copy-pasted in several places, and might be handy to make a helper function starting with the second one (I think having the full steps laid out in the first part might be helpful to newcomers).

If you do make this function, or otherwise, it would be good for each of these logger info printouts to say something unique about the step they are executing. For example, in this section it's planning between two predefined states.